### PR TITLE
DOC: update release notes for 0.18.0

### PR DESCRIPTION
**Pull request by ev-br**
[ci skip]

Please check and update.
This surely under represents all the things which were going on in `scipy.signal`, also possibly `scipy.sparse`.
Some copyediting of my ESL would be appreciated too, if anybody has an eye for it.

### CODE REVIEW

File: doc/release/0.18.0-notes.rst [hunk 1]
```diff
@@ -65,18 +70,124 @@ zero-phase low-pass FIR filter, and downsamples using `scipy.signal.upfirdn`.
...
+`scipy.special.gammaln`. For complex input, it has more consistent behavior in
+the complex plane and should be preferred over ``gammaln``.
+
+Vectorized forms of spherical Bessel functions have been implemented as 
+`scipy.special.spherical_jn`, `scipy.special.spherical_kn`,
+`scipy.special.spherical_in` and `scipy.special.spherical_yn`.
+They are recommended for use over ``sph_*`` functions, which are now deprecated.
+
+Several special functions have been extended to the complex domain and/or
+has seen domain/stability improvements. This includes `spense`, `digamma`,
```
Comment (person142) [inline doc/release/0.18.0-notes.rst]: spense -> spence and has seen -> have seen; other than that (FWIW) special looks good to me.

File: doc/release/0.18.0-notes.rst [hunk 2]
```diff
@@ -65,18 +70,124 @@ zero-phase low-pass FIR filter, and downsamples using `scipy.signal.upfirdn`.
...
+-----------------------------
+
+The implementation of Nelder-Mead minimization,
+`scipy.minimize(..., method="Nelder-Mead")`, obtained a new keyword,
+`initial_simplex`, which can be used to specify the initial simplex for the
+optimization process.
+
+Initial step size selection in CG and BFGS minimizers has been improved. We
+expect that this change will improve numeric stability of optimization in the
+vast majortiy of use cases. See pull request #5536 for details.
```
Comment (rgommers) [inline doc/release/0.18.0-notes.rst]: typo, ``majority``

Comment (rgommers) [inline doc/release/0.18.0-notes.rst]: and nitpick: ``gh-5536`` instead of ``#5536``

Comment (rgommers) [inline doc/release/0.18.0-notes.rst]: and isn't "vast majority" a bit of an overstatement? numerical stability is already fine for most usages right?

File: doc/release/0.18.0-notes.rst [hunk 3]
```diff
@@ -65,18 +70,124 @@ zero-phase low-pass FIR filter, and downsamples using `scipy.signal.upfirdn`.
...
+
+Handling of infinite bounds in SLSQP optimization has been improved. We expect
+that this change will improve numeric stability of optimization in the vast
+majortiy of use cases. See pull request #6024 for details.
```
Comment (rgommers) [inline doc/release/0.18.0-notes.rst]: same typo and comments here

File: doc/release/0.18.0-notes.rst [hunk 4]
```diff
@@ -65,18 +70,124 @@ zero-phase low-pass FIR filter, and downsamples using `scipy.signal.upfirdn`.
...
+drivers are ``gesdd`` (default) and ``gesvd``.
+
+`scipy.linalg.lapack.ilaver` returns the version of the LAPACK library SciPy
+links to.
+
+
+`scipy.spatial` improvements
+----------------------------
+
+Boolean distances, `pdist`, has been sped up. Improvements vary by the function
```
Comment (rgommers) [inline doc/release/0.18.0-notes.rst]: double backticks, this won't resolve as a link. (or `` `scipy.spatial.pdist` `` for a link)

File: doc/release/0.18.0-notes.rst [hunk 5]
```diff
@@ -65,18 +70,124 @@ zero-phase low-pass FIR filter, and downsamples using `scipy.signal.upfirdn`.
...
+
+New clustering algorithm, the nearest neighbor chain algorithm, has been
+implemented for `scipy.cluster.hierarchy.linkage`. As a result, one can expect
+a significant algorithmic improvement for several linkage methods. 
+
+
+`scipy.special` improvements
+----------------------------
+
+`scipy.special.loggamma` computes the principal branch of the logarithm of the
```
Comment (rgommers) [inline doc/release/0.18.0-notes.rst]: add that ``loggamma`` is new in 0.18.0. Preferably also with ``.. versionadded:: 0.18.0`` in the function docstring, I see that that's missing now.

**Comment by rgommers** [comment]
Content and second-language phrasing looks fine to me:)

Maybe easier to merge this - I can send a new PR on top of that for ``signal`` and a few other things.

**Comment by ev-br** [comment]
Rebased so this is mergeable again. While I was at it, addressed the review comments.

**Comment by ev-br** [comment]
> Maybe easier to merge this - I can send a new PR on top of that for signal and a few other things.

That'd be great!

**Comment by rgommers** [comment]
Great, merged.

**Pull request merged**